docs: add description and examples for docker -H (#4260)
docker/cli: docs/reference/commandline/cli.md

## Examples

### <a name="host"></a> Specify daemon host (-H, --host)

> You can use the `-H`, `--host` flag to specify one or more sockets to use when

Reviewer: I don't think "one or more" works here (for the CLI); only works on the daemon (to make it _listen_ on multiple 😅)

So.. hm.. good find actually; looks like we never updated the flag description (I think it was that way because previously that flag was shared between the daemon and the cli);

```
-H, --host list          Daemon socket(s) to connect to
```

Author: I updated the CLI help as well, removing the plural

you invoke a `docker` command. You can use the following protocols:

- `unix://`
- `tcp://`

> - `ssh://`

Reviewer: There's also `npipe://` on Windows. Perhaps we should use a table for this, something like;

```markdown
| Scheme                                 | Description               | Example                          |
|----------------------------------------|---------------------------|----------------------------------|
| `unix://[<path>]`                      | Unix socket (Linux only)  | `unix:///var/run/docker.sock`    |
| `tcp://[<IP or host>[:port]]`          | TCP connection            | `tcp://174.17.0.1:2376`          |
| `ssh://[username@]<IP or host>[:port]` | SSH connection            | `ssh://user@192.168.64.5`        |
| `npipe://[<name>]`                     | Named pipe (Windows only) | `npipe:////./pipe/docker_engine` |
```

| Scheme                                 | Description               | Example                          |
|----------------------------------------|---------------------------|----------------------------------|
| `unix://[<path>]`                      | Unix socket (Linux only)  | `unix:///var/run/docker.sock`    |
| `tcp://[<IP or host>[:port]]`          | TCP connection            | `tcp://174.17.0.1:2376`          |
| `ssh://[username@]<IP or host>[:port]` | SSH connection            | `ssh://user@192.168.64.5`        |
| `npipe://[<name>]`                     | Named pipe (Windows only) | `npipe:////./pipe/docker_engine` |

Perhaps we also need to mention what the defaults are, and that only the scheme is required when setting it as option;

- if nothing is specified, the default is `unix:///var/run/docker.sock` on Linux and macOS, and `npipe:////./pipe/docker_engine` on Windows
- `-H tcp://` uses default `tcp://localhost:2375` (or `tcp://localhost:2376` if TLS options are set)
- `-H unix://`  uses default `unix:///var/run/docker.sock`
- `-H npipe://` uses default  `npipe:////./pipe/docker_engine`

Author: Good idea, I added this

The following example shows how to invoke `docker ps` over TCP, to a remote
daemon with IP address `174.17.0.1`, listening on port `3131`:

```console

> $ docker -H tcp://174.17.0.1:3131 ps

Reviewer: perhaps use `:2376` here to encourage using the secure connection.

We have 3 ports registered; https://www.iana.org/assignments/service-names-port-numbers/service-names-port-numbers.xhtml?search=docker

- `2375`: tcp, non-TLS (insecure)
- `2376`: tcp (TLS)
- `2377`: used for swarm (secure)

So only the first 2 are relevant for this one, but perhaps we need to mention them somewhere

Author: I added a small note about it here, but we might want to add a similar description for this somewhere else too. Let's take it in a followup. LMK if the note I added seems OK to you.

```

When you use SSH invoke a command on a remote daemon, the request gets forwarded
to the `/var/run/docker.sock` Unix socket on the SSH host.

> ```

Reviewer: missing `console` code-hint here 😅

$ docker -H ssh://user@192.168.64.5 ps
```

> To achieve a similar effect without having to specify the `-H` flag for every
> command, could also [create a context](context_create.md),
> or alternatively, use the
> [`DOCKER_HOST` environment variable](#environment-variables).

Reviewer: missing "you" before `could also`.

Perhaps aa more active voice would work;

```suggestion
You can [create a context](context_create.md), or alternatively, use the
[`DOCKER_HOST` environment variable](#environment-variables) to achieve
a similar effect without having to specify the `-H` flag for every command.
```

For more information about the `-H` flag, see

> [Daemon socket option](dockerd.md#daemon-socket-option)

Reviewer: ```suggestion
[Daemon socket option](dockerd.md#daemon-socket-option).
```
